### PR TITLE
Lowercase all names passed into `StructuredLinkableSpecName`

### PR DESCRIPTION
**Pull request by courtneyholcomb**
Per title - lowercase all inputs to avoid errors later.

### CODE REVIEW

File: metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py [hunk 1]
```diff
@@ -29,6 +27,18 @@ class StructuredLinkableSpecName:
...
+    def __init__(  # noqa: D107
+        self,
+        entity_link_names: Tuple[str, ...],
+        element_name: str,
+        time_granularity_name: Optional[str] = None,
+        date_part: Optional[DatePart] = None,
+    ) -> None:
+        self.entity_link_names = tuple([entity_link_name.lower() for entity_link_name in entity_link_names])
+        self.element_name = element_name.lower()
+        self.time_granularity_name = time_granularity_name.lower() if time_granularity_name else None
```
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py:39]: Careful, the time_granularity_name can be defined in an enum, if we alter it then calling TimeGranularity(time_granularity_name) won't work. More importantly, the checks we have elsewhere to see if a given granularity is custom or not won't work for those values either.

Let's move the assertion about is_standard_name on the TimeGranularity enumeration into this initializer (unless there's a way to do that assertion in the enum class itself). Then we can delete the other dataclass elements.

File: metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py [hunk 2]
```diff
@@ -29,6 +27,18 @@ class StructuredLinkableSpecName:
...
+        self.date_part = date_part
+
     def __post_init__(self) -> None:
```
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py:42]: This is a dataclass construct and will no longer be called. Let's merge the remaining bits into `__init__` and then remove this method.

File: metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py [hunk 3]
```diff
@@ -29,6 +27,18 @@ class StructuredLinkableSpecName:
     time_granularity_name: Optional[str] = None
     date_part: Optional[DatePart] = None
```
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py:28]: Let's remove these and define property accessor methods instead.

**Review by tlento — APPROVED** [review]
I'm approving to unblock the various cross-repo wrangling we're trying to get in place, but this change needs a follow-up as it introduces a bunch of subtle issues with this class that will crop up over time. It's up to you whether to do this before merge or separately.

File: metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py (hunk 2 above)
Comment (courtneyholcomb) [reply metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py:42]: Oooh TIL!! I thought this applied to all Python classes!

**Comment by tlento** [comment]
Actually ignore the enum validation stuff and just delete post init. I found out how to verify enum structure so we'll fix that in DSI.

File: metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py (hunk 3 above)
Comment (courtneyholcomb) [reply metricflow-semantics/metricflow_semantics/naming/linkable_spec_name.py:28]: Oops that was just an oversight 🤦‍♀️ We shouldn't need to add property methods, having public attributes in the init should cover it.

**Pull request merged**